MyCryptoHQ/MyCrypto
# Add RSK network w/ network agnostic refactors

Replaces #1892 and closes #1937.

### Description

This pull request is a refactor of #1892 to be done in a more redux-y way. Rather than passing a chainId around everywhere and putting it into util functions, this implements selector functions for grabbing validation and checksum functions. Components that need to use these functions just get them as props from selectors. Would like some feedback about this approach, as I think it will come in handy as more networks differ. But maybe it's unintuitive or cumbersome, so let me know.

To simplify the address book, this also refactors it to store all addresses as lower-case'd, and only checksum them at render. In general we should always checksum at render time, and do comparisons as lower case.

### Changes

* Create selectors `getChecksumAddressFn` and `getIsValidAddressFn` and use those where necessary.
* Rework addressBook to lowercase addresses whenever storing or comparing them (cc @ConnorBryan)
* Adds all of the RSK info from #1892 (cc @alepc253)

### Steps to Test

1. Connect to RSK network.
2. Verify that ETH addresses are rightly checksummed.
3. Verify that RSK addresses are rightly checksummed.
4. Verify that an ETH address doesn't work in RSK and vice versa.
